consensus/spos/worker.go

```diff
@@ -439,8 +469,9 @@ func (wrk *Worker) checkChannels() {
 			msgType := consensus.MessageType(rcvDta.MsgType)
 			if callReceivedMessage, exist := wrk.receivedMessagesCalls[msgType]; exist {
 				if callReceivedMessage(rcvDta) {
-					if len(wrk.consensusStateChangedChannel) == 0 {
-						wrk.consensusStateChangedChannel <- true
+					select {
+					case wrk.consensusStateChangedChannel <- true:
+					default:
```

Comment:
timeout wait ?

Reply:
The channel is buffered with size=1. Actually the behavior of this code is: just to signal that an event happened. If the channel is not empty it is ok, as the notification would be send anyway so I do not need to stay here anymore